oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -953,7 +935,11 @@ impl Runtime {
         node_id: NodeId,
         half: &ChannelHalf,
     ) -> Result<ChannelReadStatus, OakStatus> {
-        self.validate_can_read_from_channel(node_id, half)?;
+        if let Err(OakStatus::ErrPermissionDenied) =
+            self.validate_can_read_from_channel(node_id, half)
+        {
+            return Ok(ChannelReadStatus::PermissionDenied);
```

Comment:
Update method documentation.

Also could you clarify when an overall `Err` is returned (if at all)?